### PR TITLE
fix methods and cleanup

### PR DESCRIPTION
**Pull request by Christian-Paul**
Note: Please ignore the failing travis build - a future PR will fix the unit tests and broken build

This PR:
- Removes content we no longer need
- Adds `itunes_type` to `to_dict` method
- Adds dedupe and subcategories logic to `add_itunes_categories`

### CODE REVIEW

**Comment by lbedner** [comment]
Was this done by a Java guy? Glad you changed all of that.

**Review by jnguyengit — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -57,3 +57,5 @@ docs/_build/
 
 # PyBuilder
 target/
+
+.idea
```
Comment (boguth) [inline .gitignore]: Newline here.

File: .travis.yml [hunk 2]
```diff
@@ -1,10 +1,35 @@
...
+deploy:
+  provider: script
+  script: bash ./scripts/deploy.sh $TRAVIS_TAG
+  on:
+    tags: true
+    condition: $TRAVIS_TAG =~ ^v[0-9]+\.[0-9]+\.[0-9]+
+# Notify us about build status.
+notifications:
+  slack:
+    secure: "Abt19+xgBqxtEXxJiDxyLRczsD8Dazdt2ONfQlWHxzpQg7ZxO8QcGXFp+pQIn+sC5pm5aB+ckN00w63tTFr6rLK8FnV8qUz6aQKIvo238B2i0kRGT9LzEGDxI2Obg+o2WNi6j5GFDzYPujufa483ELJGGUAs7WN+p82VHqd2OOGGc8VjDFCQKLCQCspGPaE/V0vEO4wXPJ/Y6pQ70c8jIhO16lecQhlZV2ROKEcf/6STS4eaWsuvoEmP/+zkyrbFD2eKwKqM+2NpE2IBOJDyVH6mNcrYbm9iYRPL1JsN1C6vACieNj5hWVtQwhCOsFQ7thtVufJVusrL7VXaKhUXPF49GDFDw6Ha1hTfUhHSq61hjWbwh6kuQqVOenyDJqMfeHcZT0NNJAuK9e0xXxkdXqOgojkx0XT9vOyU7srMvxYgnX1gGSPYOvIHuToggk16+DI7dhDP9K/vYa18Msl8KNF8kbDft1JZ4phdvDd5LvcZXxpu40QayD2SNrI8VSothpJYMq9soLjQGjKXaPnD4vkS27uc97mxDyfuD9pe33qc15JX5D75/BZDNxR8QpZUcOiuctoL+bTD56NAHPUQQon6GNnOE2rEi49nqKL63XcTUSOlAYvohdh2RhleBPC+yzLmNwu6YYRnKqpwa0gs6qORyQFTxedsFHlYzgZCmuo="
```
Comment (boguth) [inline .travis.yml]: newline here, too

File: README.rst [hunk 3]
```diff
@@ -1,10 +1,8 @@
 ###############
-pyPodcastParser
+pypodcastparser
 ###############
 
-|pypi| |pip_monthly| |testing| |coverall| |codacy| |license|
-
-pyPodcastParser is a podcast parser. It should parse any RSS file, but it specializes in parsing podcast rss feeds. pyPodcastParser is agnostic about the method you use to get a podcast RSS feed. Most user will be comfortable with the Requests_ library.
+pypodcastparser is a podcast parser. It should parse any RSS file, but it specializes in parsing podcast rss feeds. pypodcastparser is agnostic about the method you use to get a podcast RSS feed. Most user will be comfortable with the Requests_ library.
```
Comment (boguth) [inline README.rst]: Type in the last sentence ("Most user") and it looks like there's an extra space after Requests.

File: .gitignore (hunk 1 above)
Comment (Christian-Paul) [reply .gitignore]: Added

File: README.rst (hunk 3 above)
Comment (Christian-Paul) [reply README.rst]: Fixed

**Review by boguth — APPROVED** [review]

**Pull request merged**